src/main/java/com/salesforce/dataloader/controller/Controller.java

```diff
@@ -355,29 +385,33 @@ protected void initConfig(String name, boolean isBatchMode) throws ControllerIni
             // CONFIG_DIR_PROP param is NOT provided - use user's config dir
             configDir = getUserConfigDir();
             logger.debug(String.format("OS: %s, '%s' NOT provided, setting config dir to : %s", OS_TYPE, CONFIG_DIR_PROP, configDir));
-        }
-        else {
+        } else {
             // CONFIG_DIR_PROP is provided - use provided config dir
             configDir = new File(configDirPath);
             logger.debug(String.format("OS: %s, '%s' provided, setting config dir to : %s", OS_TYPE, CONFIG_DIR_PROP, configDir));
         }
 
         // Create dir if it doesn't exist
-        boolean isMkdirSuccessful = createDir(configDir);
-        if (!isMkdirSuccessful) {
-            throw new ControllerInitializationException(Messages.getMessage(getClass(), "errorCreatingOutputDir", configDir));
+        boolean isMkdirSuccessfulOrExisting = createDir(configDir);
+        if (!isMkdirSuccessfulOrExisting) {
+            String errorMsg = Messages.getMessage(getClass(), "errorCreatingOutputDir", configDir);
+            logger.error(errorMsg);
+            throw new ControllerInitializationException(errorMsg);
         }
         appPath = configDir.getAbsolutePath();
 
         // check if the config file exists
         File configFile = new File(appPath, CONFIG_FILE);
+
         String configPath = configFile.getAbsolutePath();
+        logger.info("Looking for file in config path: " + configPath);
         if (!configFile.exists()) {
 
-            File defaultConfigFile = new File(getDefaultConfigDir(), DEFAULT_CONFIG_FILE);
-
+            File defaultConfigFile = new File(getInstalledConfigDir(), DEFAULT_CONFIG_FILE);
+            logger.debug("Looking for file in config file " + defaultConfigFile.getAbsolutePath());
```

Comment:
It occurs to me that some amount of naming confusion might be avoidable by calling the "defaultConfig.properties" file the "template" rather than the "default". For example, "templateConfigDir" is a bit less confusing than "defaultConfigDir" was. "installedConfigDir" works, though.